Explore Viam (#3186)
viamrobotics/docs: docs/get-started/control-motor.md

images: ["/icons/components/motor.svg"]
description: "Use Viam to control a motor's speed and direction in just a few steps."
authors: []
weight: 20
languages: ["python", "go", "typescript", "flutter", "c++"]
viamresources: ["motor"]
no_list: true
level: "Beginner"
date: "2024-07-31"
cost: "0"
aliases:
  - /get-started/quickstarts/control-motor/
---

This quickstart is part of a series.
If you haven't read through [Learn Viam](/get-started/) and [driven a rover](/get-started/drive-rover/), we recommend you do so before continuing.

In this guide you'll configure and control a motor.

{{< alert title="You will learn" color="tip" >}}

- How to create a machine and install viam-server
- How to configure a board and a motor
- How to control your motor with UI and code

{{< /alert >}}

{{<youtube embed_url="https://www.youtube-nocookie.com/embed/awLbMqaByrE">}}

## Requirements

You **do not** need to have a physical motor to follow this quickstart.

> If you have a motor and, if needed, motor driver to hand, you can use it.

Reviewer: ```suggestion
If you have a motor and compatible motor driver on hand, you can use them.
```
You need some sort of driver/relay/something--can't just power a motor off a board.
